<?xml version="1.0"?>
<launch>

    <!-- <arg name="world_path" default="$(find dissertation_simulation)/worlds/large_warehouse.world"/> -->
    <arg name="world_path" default="$(find dissertation_simulation)/worlds"/>
    <!-- <arg name="world_path" default="$(find dissertation_simulation)/worlds/tecnopucSim.world"/> -->
    <arg name="world_name" default="empty_world"/>
    <arg name="navigating" default="true"/>
    <arg name="run_rviz" default="true"/>
    <arg name="laser_enabled" default="$(optenv HUSKY_LMS1XX_ENABLED true)"/>
    <arg name="realsense_enabled" default="$(optenv HUSKY_REALSENSE_ENABLED true)"/>
    <arg name="urdf_extras" default="$(optenv HUSKY_URDF_EXTRAS)"/>

    <!-- <param name="robot_description" command="$(find xacro)/xacro $(find husky_description)/urdf/husky.urdf.xacro laser_enabled:=$(arg laser_enabled) realsense_enabled:=$(arg realsense_enabled) urdf_extras:=$(arg urdf_extras)"/> -->

    <include file="$(find dissertation_simulation)/launch/simulation.launch">
        <arg name="world_path" value="$(arg world_path)/$(arg world_name).world"/>
        <arg name="run_rviz" value="$(arg run_rviz)"/>
        <arg name="verbose" value="true"/>
        <arg name="navigating" value="$(arg navigating)"/>
    </include>

    <include file="$(find dissertation_simulation)/launch/map_merge.launch" unless="$(arg navigating)">
      <arg name="model" value="husky"/>
    </include>

    <arg name="robot1_name" value="husky1"/>
    <group ns="$(arg robot1_name)">
        <param name="tf_prefix" value="$(arg robot1_name)_tf"/>
        <param name="/$(arg robot1_name)/map_merge/init_pose_x" value="0.00"/>
        <param name="/$(arg robot1_name)/map_merge/init_pose_y" value="0.00"/>
        <param name="/$(arg robot1_name)/map_merge/init_pose_z" value="0.00"/>
        <param name="/$(arg robot1_name)/map_merge/init_pose_yaw" value="0.00"/>
        <include file="$(find dissertation_simulation)/launch/husky_bringup.launch">
            <arg name="model" value="$(arg robot1_name)"/>
            <arg name="init_pose" value="-x 0.00 -y 0.00 -Y 0.0"/>
            <arg name="ipx" value="0.00"/>
            <arg name="ipy" value="0.00"/>
            <arg name="navigating" value="$(arg navigating)"/>
            <arg name="id_cont" value="0"/>
            <arg name="port" value="3128"/>
        </include>
    </group>

    <!-- <arg name="robot2_name" value="husky2"/>
    <group ns="$(arg robot2_name)">
        <param name="tf_prefix" value="$(arg robot2_name)_tf"/>    
        <param name="/$(arg robot2_name)/map_merge/init_pose_x" value="5.00"/>
        <param name="/$(arg robot2_name)/map_merge/init_pose_y" value="0.00"/>
        <param name="/$(arg robot2_name)/map_merge/init_pose_z" value="0.00"/>
        <param name="/$(arg robot2_name)/map_merge/init_pose_yaw" value="0.00"/>
        <include file="$(find dissertation_simulation)/launch/husky_bringup.launch">
            <arg name="model" value="$(arg robot2_name)"/>
            <arg name="init_pose" value="-x 5.00 -y 10.00 -Y 0.00"/>
            <arg name="ipx" value="7.00"/>
            <arg name="ipy" value="0.00"/>
            <arg name="navigating" value="$(arg navigating)"/>
            <arg name="id_cont" value="1"/>
            <arg name="port" value="3128"/>
        </include>
    </group>
     -->

</launch>